routes/libraries.js

```diff
@@ -239,150 +256,180 @@ router.route('/:id/survey')
      * }
      * This is using the error checking from before, using the helper functions for uniformity.
      */
+    // Grab all of the inputs from the request body.
+    let genresInput = [
+      req.body.pictureBooks,
+      req.body.youngAdultFiction,
+      req.body.fantasyFiction,
+      req.body.fairyTale,
+      req.body.boardBook,
+      req.body.nonFiction,
+      req.body.mystery,
+      req.body.graphicNovel,
+      req.body.chapterBooks,
+    ];
+
+    // For every value, if it does not exist, then the checkbox was not selected.
+    genresInput = genresInput.filter((genre) => {
+      return typeof genre === "string";
+    });
+
     try {
       updateData.fullness = validation.isValidNumber(
-        updateData.fullness,
+        parseInt(updateData.fullness),
         "Fullness Rating"
       );
-      updateData.genres = validation.checkStringArray(
-        updateData.genres,
+      genresInput = validation.checkStringArray(
+        genresInput,
         "Genres Available"
       );
+      if (genresInput.length === 0 && updateData.fullness !== 0) {
+        throw "Must specify at least one genre for a non-empty library.";
+      }
     } catch (e) {
-      return res.status(400).json({ error: e });
+      return res
+        .status(500)
+        .render("error", { errorCode: 500, title: "Error Page" });
     }
     // At this point, assume the form data is completely valid.
     try {
-      const { fullness, genres } = updateData;
       const updatedLibrary = await libraryData.formUpdate(
-        libraryId,
-        fullness,
-        genres
+        id,
+        updateData.fullness,
+        genresInput
       );
       res.json(updatedLibrary);
     } catch (e) {
-      res.status(500).render('error', {errorCode: 500, title: "Error Page"});
+      res.status(500).render("error", { errorCode: 500, title: "Error Page" });
     }
   });
 
-router.route('/:id/comments')
-  .post(async (req, res) => {
-    // Creates a new comment
-    let id;
-    
-    // If the library ID is not valid, render the error page with a status code of 400
-    try {
```

Comment:
Not sure if all this stuff is still needed

Reply:
Not sure either, kept it for now. Can be removed if needed, just a failsafe in case they try to enter a page id that's invalid.